GitCliUtil.go

```diff
@@ -28,9 +28,9 @@ func (impl *GitUtil) Fetch(rootDir string, username string, password string) (re
 	return output, "", nil
 }
 
-func (impl *GitUtil) Checkout(rootDir string, branch string) (response, errMsg string, err error) {
+func (impl *GitUtil) Checkout(rootDir string, checkoutCommit string) (response, errMsg string, err error) {
```

Comment:
checkout in checkoutCommit is redundant.. most probably its commitHash hence name should be commitHash

Reply:
This can contain branchName or tageName as well (not commit hash). Hence commitHash/commit won't make sense. 
So I am changing to only "checkout"

Reply:
checkout is already in function name, make it branchOrTagName

Reply:
It can be hash as well in branch_fix case : )
Changing to branchOrTagOrHash is not ideal.